From 901026cdb2a689ebc67f33f0c2bd070599fbcb32 Mon Sep 17 00:00:00 2001
From: Angel Krastev <akrastev@mm-sol.com>
Date: Tue, 5 Oct 2021 12:51:11 +0300
Subject: [PATCH 067/109] media: keembay-camera: Increasing isp max supported
 formats

Increased isp max supported formats and added a check so that we
no longer write out of bounds while populating isp pad formats.

Signed-off-by: Angel Krastev <akrastev@mm-sol.com>
---
 drivers/media/platform/kmb-camera/kmb-isp.c | 3 +++
 drivers/media/platform/kmb-camera/kmb-isp.h | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/kmb-camera/kmb-isp.c b/drivers/media/platform/kmb-camera/kmb-isp.c
index 9c937e80bc8c..40b710a677fe 100644
--- a/drivers/media/platform/kmb-camera/kmb-isp.c
+++ b/drivers/media/platform/kmb-camera/kmb-isp.c
@@ -453,6 +453,9 @@ static int kmb_isp_populate_pad_formats(struct kmb_isp *kmb_isp)
 				mbus.index;
 	}
 
+	if (WARN_ON(in_fmt_cnt > KMB_ISP_MAX_SUPPORTED_FMTS))
+		return -ERANGE;
+
 	for (isp_pad_idx = KMB_ISP_SRC_PAD_VID_BASE;
 		isp_pad_idx < KMB_ISP_PADS_NUM; isp_pad_idx++) {
 		for (out_fmt_cnt = 0; out_fmt_cnt < in_fmt_cnt;) {
diff --git a/drivers/media/platform/kmb-camera/kmb-isp.h b/drivers/media/platform/kmb-camera/kmb-isp.h
index c76137aeaa57..c256497ff20a 100644
--- a/drivers/media/platform/kmb-camera/kmb-isp.h
+++ b/drivers/media/platform/kmb-camera/kmb-isp.h
@@ -35,7 +35,7 @@
 /* Predefined event queue length */
 #define KMB_ISP_EVT_Q_LEN	8
 
-#define KMB_ISP_MAX_SUPPORTED_FMTS	10
+#define KMB_ISP_MAX_SUPPORTED_FMTS	20
 
 /**
  * struct kmb_isp_channel_ops - xLink channel ID operations
-- 
2.25.1

